src/stories/useAnchoredPosition.stories.tsx

```diff
@@ -0,0 +1,246 @@
+export const UseAnchoredPosition = (args: any) => {
+  const {floatingElementRef, anchorElementRef, position} = useAnchoredPosition(
+    {
+      side: `${args.anchorPosition ?? 'outside'}-${args.anchorSide ?? 'bottom'}` as AnchorSide,
+      align: args.anchorAlignment ?? 'first',
+      anchorOffset: args.anchorOffset && (parseInt(args.anchorOffset, 10) ?? undefined),
+      alignmentOffset: args.alignmentOffset && (parseInt(args.alignmentOffset, 10) ?? undefined),
+      allowOutOfBounds: args.allowOutOfBounds ?? undefined
+    },
+    [args]
+  )
+  return (
+    <Position position="absolute" top={2} bottom={0} left={2} right={0}>
+      <Anchor
+        top={args.anchorY ?? 0}
+        left={args.anchorX ?? 0}
+        width={args.anchorWidth}
+        height={args.anchorHeight}
+        ref={anchorElementRef as React.RefObject<HTMLDivElement>}
+      >
+        Anchor Element
+      </Anchor>
+      <Float
+        top={position?.top ?? 0}
+        left={position?.left ?? 0}
+        width={args.floatWidth ?? 150}
+        height={args.floatHeight ?? 150}
+        ref={floatingElementRef as React.RefObject<HTMLDivElement>}
+      >
+        Floating element
+      </Float>
+    </Position>
+  )
+}
+export const CenteredOnScreen = () => {
+  const {floatingElementRef, anchorElementRef, position} = useAnchoredPosition({
+    side: 'inside-center',
+    align: 'center'
+  })
+  // The outer Position element simply fills all available space
+  return (
+    <Position
+      ref={anchorElementRef as React.RefObject<HTMLDivElement>}
+      position="absolute"
+      top={0}
+      bottom={0}
+      left={0}
+      right={0}
+    >
+      <Float
+        ref={floatingElementRef as React.RefObject<HTMLDivElement>}
+        top={position?.top ?? 0}
+        left={position?.left ?? 0}
+      >
+        <p>Screen-Centered Floating Element </p>
+        <p>
+          <small>
+            <em>(Controls are ignored for this story)</em>
+          </small>
+        </p>
+      </Float>
+    </Position>
+  )
+}
+
+const Nav = styled('nav')`
+  width: 300px;
+  padding: ${get('space.3')};
+  position: relative;
+  overflow: hidden;
+  border-right: 1px solid ${get('colors.border.gray')};
+`
+const Main = styled('main')`
+  display: flex;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+`
+
+/*
+
+There are a few "gotchas" to take note of from this example. See the
+documentation for more info.
+
+1. The portal's root (<Main> in this example) needs to be large enough
+   to include ANY space that the overlay might need to take. By default,
+   elements are not rendered at full height! Notice how <Main> uses
+   top, left, right, and bottom all set to 0 to achieve a full-size box.
+
+2. The positioning routine needs to know the size of the overlay before
+   calculating its position! Therefore, we use visibility: hidden to
+   prevent showing a single frame of the overlay being positioned at
+   (0, 0).
+
+*/
```

Comment:
Can you call out both of these in the docs? You've covered the first one already but might be worth mentioning this is something you'll need to be even more careful about if using a Portal